packages/playground/src/components/node-listener/node-listener.tsx

```diff
@@ -0,0 +1,136 @@
+declare var commonTypes;
+
+import { Component } from "@stencil/core";
+
+import CounterfactualNode from "../../data/counterfactual";
+import FirebaseDataProvider from "../../data/firebase";
+import { WidgetDialogSettings } from "../../types";
+
+// TODO: This should be imported from @counterfactual/common-types.
+const { MethodName } = commonTypes.Node;
+
+type NodeMessageHandlerCallback = (data: any) => void;
+type NodeMessageResolver = { [key: string]: NodeMessageHandlerCallback };
+
+@Component({
+  tag: "node-listener",
+  shadow: true
+})
+export class NodeListener {
+  private currentMessage: string = "";
+
+  private modals: { [key: string]: (data: any) => WidgetDialogSettings } = {
+    proposeInstall: data => ({
+      content: (
+        <label>
+          You're about to deposit <strong>${data.eth}</strong>
+          to play <strong>${data.appName}</strong>
+          with <strong>${data.peerName}</strong>
+        </label>
+      ),
+      primaryButtonText: "Accept",
+      secondaryButtonText: "Reject",
+      onPrimaryButtonClicked: this.acceptProposeInstall.bind(this, data),
+      onSecondaryButtonClicked: this.rejectProposeInstall.bind(this, data)
+    })
+  };
+
+  private nodeMessageResolver: NodeMessageResolver = {
+    [MethodName.PROPOSE_INSTALL]: this.handleProposeInstall.bind(this),
+    [MethodName.REJECT_INSTALL]: this.handleRejectInstall.bind(this)
+  };
+
+  private modalVisible: boolean = false;
+  private modalData: WidgetDialogSettings = {} as WidgetDialogSettings;
+
+  get node() {
+    return CounterfactualNode.getInstance();
+  }
+
+  private get currentModalConfiguration():
+    | ((data: any) => WidgetDialogSettings)
+    | null {
+    if (this.currentMessage) {
+      return this.modals[this.currentMessage];
+    }
+
+    return null;
+  }
+
+  componentWillLoad() {
+    // TODO: This is a non-existant firebase data provider.
+    const serviceProvider = new FirebaseDataProvider({
+      apiKey: "AIzaSyBne_N_gQgaGnyfIPOs9T0PhOPdwRUeUsI",
+      authDomain: "joey-firebase-1.firebaseapp.com",
+      databaseURL: "https://joey-firebase-1.firebaseio.com",
+      projectId: "joey-firebase-1",
+      storageBucket: "joey-firebase-1.appspot.com",
+      messagingSenderId: "86354058442"
+    });
```

Comment:
Might be good to have these values flow in from an `.env` file

Reply:
@snario This should actually come from the Playground Server backend. We can't use `.env` files for browser-facing apps.